### PR TITLE
Table Improvements

### PR DESCRIPTION
**Pull request by jpjsonic**
In this PR

- [x] Download trace cohort data as a csv
- [x] Add global search bar & improve column filter functionality 
- [x] Add rows per page option
- [x] Enable grouping Column aggregations when using `column_defs`

### CODE REVIEW

File: viewer/package.json [hunk 1]
```diff
@@ -40,7 +41,7 @@
   },
   "scripts": {
     "start": "REACT_APP_API_BASE_URL='http://localhost:8000' react-scripts start",
-    "test": "react-scripts test",
+    "test": "react-scripts test --transformIgnorePatterns \"node_modules/(?!export-to-csv)/\"",
```
Comment (tgsoverly) [inline viewer/package.json:44]: This can go in the "jest" block.  See my dag PR for an example.

File: visivo/models/table.py [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
 from .base.parent_model import ParentModel
 from .base.base_model import REF_REGEX, generate_ref_field
 from pydantic import model_validator
+from enum import IntEnum
+
+class RowsPerPageEnum(IntEnum):
+    three = 3
+    five = 5
+    ten = 15
+    twentyfive = 25
```
Comment (tgsoverly) [inline visivo/models/table.py:17]: twenty_five etc.

File: viewer/src/components/items/Table.js [hunk 3]
```diff
@@ -1,100 +1,149 @@
...
         }
     }, [selectedTableCohort, tracesData, table]);
 
     useEffect(() => {
         if (selectedTableCohort && columns) {
-            setTableData(tableDataFromCohortData(selectedTableCohort.data, columns))
+            setTableData(tableDataFromCohortData(selectedTableCohort.data, columns));
         }
     }, [selectedTableCohort, columns]);
+    const handleExportData = () => {
```
Comment (tgsoverly) [inline viewer/src/components/items/Table.js:47]: Wonder if this can be tested.

**Review by tgsoverly — APPROVED** [review]

File: viewer/package.json (hunk 1 above)
Comment (jpjsonic) [reply viewer/package.json:44]: Cool, will move it there 👍

File: viewer/src/components/items/Table.js (hunk 3 above)
Comment (jpjsonic) [reply viewer/src/components/items/Table.js:48]: Added a test for clicking the download button!

**Pull request merged**